File: app/src/main/java/com/hgnis/reader/views/activity/SplashScreen.java
package com.hgnis.reader.views.activity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.hgnis.reader.R;

public class SplashScreen extends AppCompatActivity {

    Intent i;
    Handler hand;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_screen);
        hand = new Handler();


    }

    @Override
    protected void onStart() {
        super.onStart();
        hand.postDelayed(new Runnable() {
            @Override
            public void run() {
                i = new Intent(SplashScreen.this, MainActivity.class);
                startActivity(i);
                finish();
            }
        }, 1500);
    }
}

File: app/src/main/java/com/hgnis/reader/views/activity/SplashScreen.kt
package com.hgnis.reader.views.activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import com.hgnis.reader.R
import com.hgnis.reader.utility.AppSharePreference


class splashScreen : AppCompatActivity() {
    var appSharePreference: AppSharePreference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        appSharePreference = AppSharePreference(this)
        if (appSharePreference!!.name.isNotEmpty()) {
            Handler(Looper.getMainLooper()).postDelayed({
                val intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
                finish()
            }, 1500)
        } else {
            val intent = Intent(this, login::class.java)
            startActivity(intent)
            finish()
        }

    }
}